wall_e_install.sh

```diff
@@ -1,34 +1,19 @@
 #!/usr/bin/env bash
-set -e
-
 echo "installing ESP IDF"
 
 _shell_="${SHELL#${SHELL%/*}/}"
-
-_sudo(){
-    if ! command -v sudo &> /dev/null
-    then
-        if ! [ $(id -u) = 0 ]; then
-           echo "The script need to be run as root." >&2
-           exit 1
-        fi
-        "$@"
-    else
-        sudo "$@"
-    fi
-}
-
 # Check whether esp-idf has already been installed
 if [ -d $HOME/esp/esp-idf ]; then
-    echo "You have installed esp-idf" && exit 0
-fi
-
-unameOut="$(uname -s)"
+    echo "You have already installed esp-idf!"
+else
+    # System Detection and ESP-IDF Installation
+    unameOut="$(uname -s)"
 case "${unameOut}" in
     Linux*)
-        export DEBIAN_FRONTEND=noninteractive
-        _sudo apt update && _sudo apt upgrade -y
-        _sudo apt install git wget flex bison gperf python3 python3-pip python3-setuptools cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0 -y
+        sudo apt update && sudo apt upgrade -y
```

Comment:
So, when you removed the `_sudo` function, this will ask for password here. The script is supposed to run without interruptions once started, so why remove the function?